### ISSUE TITLE
Add support for using @CheckedChange on a RadioGroup

### ISSUE THREAD

**Issue by alan-mee** [excilys/androidannotations#1309]
I think it would be useful if @CheckChange could also work on a RadioGroup.

Right now I get "Error:(108, 49) error: incompatible types: <anonymous android.widget.CompoundButton.OnCheckedChangeListener> cannot be converted to android.widget.RadioGroup.OnCheckedChangeListener" when trying to use @CheckedChange on a RadioGroup.  There are prior issues that have been created regarding this, but all have been closed.

https://github.com/excilys/androidannotations/issues/389 - Abandoned?
https://github.com/excilys/androidannotations/issues/1086 - Requester was not clear?
https://github.com/excilys/androidannotations/issues/1004 - Workaround to use events for each button.

**Comment by WonderCsabo** [comment]
I have only one problem with that: it is crucial to have the `int checkedId` parameter in case of a `RadioGroup`, but useless in case of `CompoundButton`s. So we may mix different things into one annotation. On the contrary, i would not add a new annotation for this.

@yDelouis what do you think?

**Comment by WonderCsabo** [comment]
@yDelouis @dodgex any thoughts?

**Comment by dodgex** [comment]
From what i have seen we either need a new annotation or a parameter to toggle usage of the RadioGroup listener as there is no way to detect the correct type at compile time.

or we could decide based on the method parameters... if int present then do radiogroup. but that is... meh.

**Comment by WonderCsabo** [comment]
I agree,  this would not be clear by adding it to the existing annotation.

**Comment by Rainer-Lang** [comment]
any news?

**Comment by WonderCsabo** [comment]
We currently not working on this, because we are occupied with other
issues. But we welcome any contributions!

**Comment by gauravsak** [comment]
@WonderCsabo @dodgex @yDelouis If anyone wants to implement this, what is the final guideline about the API?

**Comment by WonderCsabo** [comment]
If the first method parameter is `RadioGroup`, then we should generate the `RadioGroup` listener, otherwise we should assume it is a `CompoundButton`.

**Comment by dodgex** [comment]
After an [attempt to support](https://github.com/androidannotations/androidannotations/pull/1937) both `RadioGroup` and `CompoundButton` with `@CheckedChange` we decided to create a new annotation for `RadioGroup`
